### PR TITLE
Local SSL tutorial

### PR DESCRIPTION
**Pull request by nmarinel**
## Hey, I just made a Pull Request!

No code, just a contrib doc :). This was suggested after some discussion in the `auth` Discord channel about how to enable TLS for local testing.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jhaals** [comment]
Thanks for the suggestion! Perhaps these docs could be simplified by using https://github.com/FiloSottile/mkcert instead. That should remove both the generation step and possibly the need for the node environment variable as well.

**Comment by nmarinel** [comment]
@jhaals mkcert is a pretty neat tool, thanks for sharing that. I'll admit I struggled with it a bit running Windows/WSL(I first tried installing it in WSL, which Windows doesn't recognize as a CA store), but the problems weren't anything we wouldn't expect the average dev to work through. I've updated the instructions to use mkcert instead of openssl.

One unfortunate note, it looks like the [node environment variable is needed even with mkcert](https://github.com/FiloSottile/mkcert#using-the-root-with-nodejs).

**Comment by benjdlambert** [comment]
You could also maybe introduce some docs for the #13338 that's also introducing the webpack config for enabling https for the frontend locally too once that's merged? Just thinking that it would be good to align these.

**Comment by nmarinel** [comment]
Good call, @benjdlambert. I've updated the instructions for the frontend to reflect the updated config. Also I realized you can use `$file` instead of copying the contents of the file, so that deserved to be included :). 

LMK what you think.

File: contrib/docs/tutorials/enable-ssl-self-signed.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+         cert: |
+           -----BEGIN CERTIFICATE-----
+           MIIDCTCCAfGgAwIBAgIUZ9VhZckcy690L
+           ...
+           -----END CERTIFICATE-----
+         # ... or use a path
+         key:
+           $file: ./certs/localhost-key.pem
+   ```
+1. Start the backend with `NODE_EXTRA_CA_CERTS=/absolute/path/to/cert.pem yarn start-backend`
```
Comment (jhaals) [inline contrib/docs/tutorials/enable-ssl-self-signed.md:25]: is this needed if you use mkcert as it installs the certificate in the trust store? Would you be able to verify if it works without it? 🙏

**Review by jhaals — CHANGES_REQUESTED** [review]
Sorry for the wait time! This has for some reason dropped out of my review inbox. One small comment but otherwise looking good :)

**Review by jhaals — APPROVED** [review]
I just realised that you already tried without the environment variable, thanks!

**Pull request merged**